source/authentication/api-key.txt

```diff
@@ -81,123 +67,168 @@ Configuration
+to allow them to authenticate directly with App Services.
+
+You can associate up to 100 server API keys with an app.
 
 .. _api-key-authentication-usage-create-server-key:
 
 Create a Server API Key
------------------------
+~~~~~~~~~~~~~~~~~~~~~~~
+
+You must enable the API key provider before you can create an API key.
+
+You must copy the server key's value as soon as you create it.
+Once you create the API key, you can no longer retrieve it using the admin interfaces.
 
 .. tabs-realm-admin-interfaces::
-   
+
    .. tab::
       :tabid: ui
-      
+
       #. Select :guilabel:`App Users` from the {+leftnav+}.
-      
+
       #. Select the :guilabel:`Authentication Providers` button and then select
          the :guilabel:`API Keys` provider.
-      
+
       #. If the API Key provider is not currently enabled, enable it and then
          deploy your application.
-      
+
       #. Click :guilabel:`Create API Key`.
-      
+
       #. Enter a unique name for the key and then click :guilabel:`Save`.
-   
+
    .. tab::
       :tabid: cli
-      
-      .. tip::
-         
-         You must enable the API key provider before you can create an API key.
-      
+
       To create a new server API key, call ``realm-cli users create`` and
       specify ``--type=api-key``. The CLI will prompt you for your App ID as
       well as a name for the new API key.
-      
+
       .. code-block:: bash
-         
+
          realm-cli users create --type=api-key
-      
-      You can also specify the arguments when you call the program:
-      
+
+      You can also specify the arguments in the command:
+
       .. code-block:: bash
-         
+
          realm-cli users create --type=api-key \
            --app=<Your App ID> \
            --name=<API Key Name>
 
-.. important::
+   .. tab::
+      :tabid: api
 
-   Remember to copy the server key's value as soon as you create it.
-   Once you leave the provider configuration page or disable the key you
-   cannot retrieve the value from the App Services UI.
+      To create a server API key using the Admin API, make a request to the
+      `Create a new API key <{+base-url+}{+admin-api-page+}tag/apikeys/operation/adminListApiKeys>`__ endpoint.
+
+.. _api-key-authentication-user-api-key:
+
+User API Keys
+-------------
+
+You can generate user API keys with the Realm SDKs.
+User API keys are generated for specific application users by the Realm
+SDKs in client applications. You can then use the user API to authenticate as that user.
+
+You can allow devices or services to communicate with App Services on behalf
+of a user by associating a unique user API key with each device.
+
+User API keys are always associated with a non-anonymous user. Each user can
+associate up to 20 user API keys with their account.
+
+To learn how to generate user API keys, refer to the Realm SDK documentation.
+You can find the relevant documentation in the following Realm SDK Examples section.
```

Comment:
This sentence is dangerous, since we're revamping the IA of the SDKs...and they are likely to change again. In other words, there might no longer be an Examples section in any given SDK (or all of them).

Reply:
Ah, I see now. I looked at the preview. Yeah, so instead of above, I'd just get rid of the sentence, the next subsection header and blurb, and combine them to have "You can find the relevant documentation in the SDK docs for your platform: 
[Tabs]

Reply:
thx good spot. removing